### PR TITLE
Features/42-read-txt-horiba-xy

### LINKED ISSUE #42: Review, add and improve unit tests
[Codecov](https://codecov.io/gh/r-hyperspec/hySpc.read.txt?branch=develop)

Review, fix, add and improve unit tests for these functions and their helpers:

- [x] count_lines

- [x] read_asc_Andor
- [x] read_asc_PerkinElmer

- [x] read_txt_Shimadzu

- [x] #48 
- [x] #49
- [x] #72 
    - read_txt_Horiba
    - read_txt_Horiba_t
    - read_txt_Horiba_xy
    

-  [LOW PRIORITY, may skip for now] #74
    - read_ini

- [x] #70 
    - read_zip_Renishaw
    - read_txt_Renishaw

- [x] #75
    - read_dat_Witec
    - read_txt_Witec
    - read_txt_Witec_Graph
    - read_txt_Witec_TrueMatch
    

- [x] #73

### PR DESCRIPTION
**Pull request by sangttruong**
Add unittest for read_txt_Horiba_xy()
Resolve #42

### CODE REVIEW

**Comment by GegznaV** [comment]
Why are there 150+ files changed in this PR when only 1 file should be? (Most probably the files currently are in the wrong place to be reached by unit tests).
I also noticed that the same trend of 150+ changed files is in the last few PRs (which makes it more difficult to review). @sangttruong, are the same files moved again and again in separate PRs? Or are these different files?

File: R/read_txt_Horiba.R [hunk 1]
```diff
@@ -87,3 +87,59 @@ read_txt_Horiba_t <- function(file, header = TRUE, sep = "\t", row.names = NULL,
 # Unit tests -----------------------------------------------------------------
 
 # FIXME: add unit tests
```
Comment (GegznaV) [inline R/read_txt_Horiba.R]: Why is  `fixme ` left?

File: R/read_txt_Horiba.R [hunk 2]
```diff
@@ -87,3 +87,59 @@ read_txt_Horiba_t <- function(file, header = TRUE, sep = "\t", row.names = NULL,
...
+  test_that("Horiba time series .txt labels are correct",{
+    expect_true(is.expression(spc@label$.wavelength))
+    expect_true(is.expression(spc@label$spc))
+    expect_equal(spc@label$filename, "filename")
+  })
+
+  test_that("Horiba time series .txt spectral data", {
+    expect_equal(dim(spc@data$spc), c(100,1024))
+
+    expect_equal(colnames(spc@data$spc)[[974]], "5.9526")
```
Comment (cbeleites) [inline R/read_txt_Horiba.R]: `colnames()` returns a vector, please use single square brackets.
Same for the wavelength vector below.

File: R/read_txt_Horiba.R [hunk 3]
```diff
@@ -87,3 +87,59 @@ read_txt_Horiba_t <- function(file, header = TRUE, sep = "\t", row.names = NULL,
...
 # FIXME: add unit tests
+
+hySpc.testthat::test(read_txt_Horiba_t) <- function() {
+  context("read_txt_Horiba_t")
+
+  ts <- system.file("extdata/txt.HoribaJobinYvon/", "ts.txt",package = "hySpc.read.txt")
+  spc <- read_txt_Horiba_t(ts)
+
+  test_that("Horiba time series .txt labels are correct",{
+    expect_true(is.expression(spc@label$.wavelength))
```
Comment (cbeleites) [inline R/read_txt_Horiba.R]: please check via `as.character()` like you did with the JCAMP-DX tests.

File: R/read_txt_Horiba.R [hunk 4]
```diff
@@ -87,3 +87,59 @@ read_txt_Horiba_t <- function(file, header = TRUE, sep = "\t", row.names = NULL,
...
+    expect_equal(colnames(spc@data$spc)[[123]], "734.39")
+  })
+
+  test_that("Horiba map .txt wavelength", {
+    expect_equal(length(spc@wavelength), 616)
+
+    expect_equal(round(spc@wavelength[[85]], 3), 521.219)
+    expect_equal(round(spc@wavelength[[431]], 3), 2462.195)
+  })
+}
```
Comment (cbeleites) [inline R/read_txt_Horiba.R]: please also test the map dimensions x and y

File: R/read_txt_Horiba.R [hunk 5]
```diff
@@ -87,3 +87,59 @@ read_txt_Horiba_t <- function(file, header = TRUE, sep = "\t", row.names = NULL,
...
+    expect_equal(colnames(spc@data$spc)[[657]], "772.27")
+  })
+
+  test_that("Horiba time series .txt wavelength", {
+    expect_equal(length(spc@wavelength), 1024)
+
+    expect_equal(round(spc@wavelength[[79]], 3), 1986.863)
+    expect_equal(round(spc@wavelength[[1011]], 3), -88.828)
+  })
+}
```
Comment (cbeleites) [inline R/read_txt_Horiba.R:142]: please also test the time dimension/column

**Review by cbeleites — CHANGES_REQUESTED** [review]

**Comment by GegznaV** [comment]
image




@sangttruong, Did you answer or solve @cbeleites comments like the one on the print screen above?
 
 You can find them under the "Files changes" section (this applies to all pull requests).
 
image

 
In case of doubt about what to do, please, answer to the comments directly on GitHub and not on Slack so that every mentor could contribute to the discussion and could know what is the status of the issue.

File: R/read_txt_Horiba.R [hunk 6]
```diff
@@ -86,4 +86,58 @@ read_txt_Horiba_t <- function(file, header = TRUE, sep = "\t", row.names = NULL,
...
+
+    expect_equal(colnames(spc@data$spc)[974], "5.9526")
+    expect_equal(colnames(spc@data$spc)[657], "772.27")
+  })
+
+  test_that("Horiba time series .txt wavelength", {
+    expect_equal(length(spc@wavelength), 1024)
+
+    expect_equal(round(spc@wavelength[79], 3), 1986.863)
+    expect_equal(round(spc@wavelength[1011], 3), -88.828)
```
Comment (GegznaV) [inline R/read_txt_Horiba.R]: Avoid rounding, if possible

File: R/read_txt_Horiba.R (hunk 4 above)
Comment (GegznaV) [reply R/read_txt_Horiba.R]: Add this as additional unit tests

**Review by GegznaV — CHANGES_REQUESTED** [review]
@sangttruong, add unit tests defined in template #55 as additional tests to the current ones.

**Review by bryanhanson — APPROVED** [review]
As far as I can tell, all comments and requested changes have been made; if something has been missed a new issue can be opened.  OK to merge.

**Pull request merged**